### PR TITLE
Added file filter for drag & drop functionality.

### PR DESCRIPTION
**Pull request by SidaDan**
Hi again, 
I added a file filter for drag & drop functionality and fixed a little bug
Bug fixed: If modified then ask to save the changes before drop a file.

Have a nice day.

### CODE REVIEW

**Comment by pH-7** [comment]
Looks amazing 💯 @SidaDan!!

File: src/simplejavatexteditor/UI.java [hunk 1]
```diff
@@ -630,13 +643,25 @@ public void drop(DropTargetDropEvent e) {
...
                         try {
                             String fileName = tr.getTransferData(flavors[i]).toString().replace("[", "").replace("]", "");
-                            FileInputStream fis = new FileInputStream(new File(fileName));
-                            byte[] ba = new byte[fis.available()];
-                            fis.read(ba);
-                            textArea.setText(new String(ba));
-                            fis.close();
-                        }
-                        catch (Exception ex) {
+                            String[] extensionFilter = {".txt", ".dat", ".log", ".xml", ".mf", ".html"}; // allowed file filter extentions for drag and drop
```
Comment (pH-7) [inline src/simplejavatexteditor/UI.java:646]: Can this array be a class attribute instead? We might have to reuse it later on elsewhere in the class... (and easier to find it if we have to add more file extensions). What do you think?

Comment (SidaDan) [reply src/simplejavatexteditor/UI.java:646]: Hi. yes of course. Is only an example. You can create a setup.class as well.

Have a nice day

Comment (pH-7) [reply src/simplejavatexteditor/UI.java:646]: Yes, that was a suggestion 😉

**Pull request merged**